macOS/DuckDuckGoVPN/VPNNotificationsObserver.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+import VPNNotifications
+import os.log
+
+final class VPNNotificationsObserver {
+
+    private let notificationsPresenter = {
+        let parentBundlePath = "../../../../"
+        let mainAppURL: URL
+
+        if #available(macOS 13, *) {
+            mainAppURL = URL(filePath: parentBundlePath, relativeTo: Bundle.main.bundleURL)
+        } else {
+            mainAppURL = URL(fileURLWithPath: parentBundlePath, relativeTo: Bundle.main.bundleURL)
+        }
+
+        return VPNNotificationsPresenter(appLauncher: AppLauncher(appBundleURL: mainAppURL))
+    }()
+
+    private let distributedNotificationCenter = DistributedNotificationCenter.default()
+
+    // MARK: - Notifications: Observation Tokens
+
+    private var cancellables = Set<AnyCancellable>()
+
+    func startObservingVPNStatusChanges() {
+        Logger.networkProtection.log("Register with sysex")
+
+        distributedNotificationCenter.publisher(for: .showIssuesStartedNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.showReconnectingNotification()
+            }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .showConnectedNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] notification in
+                let serverLocation = notification.object as? String
+                self?.showConnectedNotification(serverLocation: serverLocation)
+            }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .showIssuesNotResolvedNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.showConnectionFailureNotification()
+            }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .showVPNSupersededNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.showSupersededNotification()
+            }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .showTestNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.showTestNotification()
+            }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .serverSelected).sink { [weak self] _ in
+            Logger.networkProtection.log("Got notification: listener started")
+            self?.notificationsPresenter.requestAuthorization()
+        }.store(in: &cancellables)
+
+        distributedNotificationCenter.publisher(for: .showExpiredEntitlementNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.showEntitlementNotification()
+            }.store(in: &cancellables)
+    }
+
+    func applicationSupportsSecureRestorableState(_ app: NSApplication) -> Bool {
+        return true
+    }
```

Comment:
Can this be removed? I don’t see where this type is set as a delegate anywhere so unsure why it’s required.